backend/infrahub/cli/patch.py

```
@@ -0,0 +1,142 @@
+
+from .constants import ERROR_BADGE, SUCCESS_BADGE
+
+if TYPE_CHECKING:
+    from infrahub.cli.context import CliContext
+    from infrahub.database import InfrahubDatabase
+
+
+patch_app = AsyncTyper(help="Commands for planning, applying, and reverting database patches")
+
+
+def get_patch_runner(db: InfrahubDatabase) -> PatchRunner:
+    plan_writer = PatchPlanWriter()
+    plan_reader = PatchPlanReader()
+    return PatchRunner(
+        plan_writer=plan_writer,
+        plan_reader=plan_reader,
+        edge_db_id_translator=PatchPlanEdgeDbIdTranslator(),
+        vertex_adder=PatchPlanVertexAdder(db=db),
+        vertex_deleter=PatchPlanVertexDeleter(db=db),
+        vertex_updater=PatchPlanVertexUpdater(db=db),
+        edge_adder=PatchPlanEdgeAdder(db=db),
+        edge_deleter=PatchPlanEdgeDeleter(db=db),
+        edge_updater=PatchPlanEdgeUpdater(db=db),
+    )
+
+
+@patch_app.command(name="plan")
+async def plan_patch_cmd(
+    ctx: typer.Context,
+    patch_path: str = typer.Argument(
+        help="Path to the file containing the PatchQuery instance to run. Use Python-style dot paths, such as infrahub.cli.patch.queries.base"
+    ),
+    patch_plans_dir: Path = typer.Option(Path("infrahub-patches"), help="Path to patch plans directory"),  # noqa: B008
+    config_file: str = typer.Argument("infrahub.toml", envvar="INFRAHUB_CONFIG"),
+) -> None:
+    """Create a plan for a given patch and save it in the patch plans directory to be applied/reverted"""
+    logging.getLogger("infrahub").setLevel(logging.WARNING)
+    logging.getLogger("neo4j").setLevel(logging.ERROR)
+    logging.getLogger("prefect").setLevel(logging.ERROR)
+
+    patch_module = importlib.import_module(patch_path)
+    patch_query_class = None
+    for _, cls in inspect.getmembers(patch_module, inspect.isclass):
+        if issubclass(cls, PatchQuery) and cls is not PatchQuery:
+            patch_query_class = cls
+            break
```

Will this always be from files that we control? De we need to raise an error if there are two classes inheriting from `PatchQuery`?

---

good point. added handling for this case and improved the error message